api/src/main/java/ca/bc/gov/educ/api/gradstudent/controller/GraduationStatusController.java

```diff
@@ -391,6 +391,22 @@ public ResponseEntity<Integer> getStudentsCountForAmalgamatedSchoolReport(@PathV
         return response.GET(gradStatusService.countStudentsForAmalgamatedSchoolReport(schoolOfRecord));
     }
 
+    @PostMapping (EducGradStudentApiConstants.STUDENT_COUNT)
+    @PreAuthorize(PermissionsConstants.READ_GRADUATION_STUDENT)
+    @Operation(summary = "Get Students Count by mincode and status", description = "Get Students Count by mincode and status", tags = { "Business" })
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "OK")})
+    public ResponseEntity<Long> getStudentsCount(@RequestParam(required = false) String studentStatus, @RequestBody List<String> schoolOfRecords) {
+        return response.GET(gradStatusService.countBySchoolOfRecordsAndStudentStatus(schoolOfRecords, studentStatus));
+    }
+
+    @PostMapping (EducGradStudentApiConstants.STUDENT_ARCHIVE)
+    @PreAuthorize(PermissionsConstants.READ_GRADUATION_STUDENT)
```

Comment:
READ_GRADUATION_STUDENT doesn't look like the right permission for archiving students.

Reply:
Updated